plugins/nf-co2footprint/src/resources/assets/CO2FootprintReportTemplate.js

```diff
@@ -77,8 +76,25 @@ $(function() {
     energy_data.push({y: norm_units(smry.energy), name: pname, type:'box', boxmean: true, boxpoints: false});
   }
 
-  Plotly.newPlot('co2eplot', co2e_data, { title: 'CO2 emission', yaxis: {title: 'CO2 emission (g)', tickformat: '.2e', rangemode: 'tozero'} });
-  Plotly.newPlot('energyplot', energy_data, { title: 'Energy consumption', yaxis: {title: 'Energy consumption (Wh)', tickformat: '.2s', rangemode: 'tozero'} });
+  // Decide yaxis tickformat
+  co2e_data.forEach(function (p) {
+    max = 0;
+    if (p !== null) {
+      max = Math.max(max, p.y);
+    }
+  });
+  var co2e_tickformat = (max <= 4) ? ('.2f') : ('.2s');
+  energy_data.forEach(function (p) {
+    max = 0;
+    if (p !== null) {
+      max = Math.max(max, p.y);
+    }
+  });
```

Comment:
Hi, when checking the output I observed that for some low values it still uses the `.2s` `tickformat`. I briefly checked the code: `max` is `NaN` (i.e. that's why the tickformat always stays `.2s`) and it seems that something is wrong with the max() assignment.

Reply:
Solved now, the problem happened when there are boxes in the plot, then `p.y` is an array

Reply:
Great, the plot for the smaller values looks good now!